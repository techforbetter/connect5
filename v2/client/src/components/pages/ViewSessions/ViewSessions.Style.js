import styled from 'styled-components';
import { Link } from 'react-router-dom';
import { colors } from '../../../theme';

export const ViewSessionsWrapper = styled.div`
  display: flex;
  flex-direction: column;
  width: 90%;
  margin: 0 auto;
    @media (min-width: 768px) {
      width: 50%
`;

export const TotalSessions = styled.div`
  width: 80%;
  margin: 0 auto;
  margin-top: 6rem;
`;

export const Span = styled.p`
  text-align: center;
`;

export const SessionsCount = styled.p`
  text-align: center;
  font-weight: 600;
  font-size: 1.5rem;
  margin-top: 1.5rem;
  margin-bottom: 1.5rem;
`;

export const LinkBtn = styled(Link)`
  background-color: ${colors.lightPrimary};
  border-radius: 15px;
  border: 1px solid ${colors.lightPrimary};
  width: 200px;
  display: block;
  text-align: center;
  margin: 0 auto;
  margin-bottom: 2rem;
  cursor: pointer;
  color: ${colors.white};
  font-size: 1rem;
  padding: 14px 25px;
  text-decoration: none;
  text-shadow: 0px 1px 0px #2f6627;
  &:hover {
	background-color: ${colors.white};
  color: ${colors.lightPrimary};
  border: 1px solid ${colors.lightPrimary}
}
  &:active {
	position:relative;
	top:1px;
`;
