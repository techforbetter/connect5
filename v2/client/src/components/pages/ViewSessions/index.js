/* eslint-disable react/destructuring-assignment */
import React, { Component } from 'react';
import { connect } from 'react-redux';
import moment from 'moment';
import { Empty } from 'antd';

// COMMON COMPONENTS
import SessionList from '../../common/List/SessionList';
import Header from '../../common/Header';
import Toggle from '../../common/Toggle';

// ACTIONS
import {
  fetchTrainerSessions,
  fetchLocalLeadSessions,
  fetchALLSessions,
} from '../../../actions/groupSessionsAction';

// STYLING
import {
  ViewSessionsWrapper,
  TotalSessions,
  Span,
  SessionsCount,
  LinkBtn,
  HeaderSection,
} from './ViewSessions.Style';

class ViewSessions extends Component {
  state = {
    headerTitle: null,
    toggle: 'left',
  };

  componentDidMount() {
    this.getData();
  }

  componentDidUpdate(prevProps) {
    const { path: oldPath } = prevProps.match;
    const { path } = this.props.match;
    if (path !== oldPath) {
      this.getData();
    }
  }

  getData = () => {
    const { userId, match, role } = this.props;
    const { localLeadId, trainerId } = match.params;
    let resultsFor;
    let resultForRule;
    let headerTitle;
    // admin || local lead || trainer viewing his own session
    if (match && match.path === '/my-sessions') {
      resultsFor = userId;
      resultForRule = 'trainer';
      headerTitle = 'Your Sessions';
      this.fetchSessionsData(resultForRule, resultsFor);
    } else if (
      (match &&
        match.path === '/group-results/:trainerId?' &&
        role === 'admin') ||
      role === 'localLead'
    ) {
      if (localLeadId) {
        resultsFor = localLeadId;
        resultForRule = 'localLead';
        headerTitle = 'Group Sessions';
        this.fetchSessionsData(resultForRule, resultsFor);
      } else {
        resultsFor = userId;
        resultForRule = 'localLead';
        headerTitle = 'Yours Group Sessions';
        this.fetchSessionsData(resultForRule, resultsFor);
      }
    } else if (match && match.path === '/all-sessions') {
      resultsFor = userId;
      resultForRule = 'admin';
      headerTitle = 'All Sessions';

      this.fetchSessionsData(resultForRule, resultsFor);
    } else if (match && match.path === '/trainer-sessions/trainerId?') {
      if (trainerId) {
        resultsFor = trainerId;
        resultForRule = 'trainer';
        headerTitle = 'Trainer Sessions';

        this.fetchSessionsData(resultForRule, resultsFor);
      }
    }
    this.setState({ headerTitle });
  };

  fetchSessionsData = (role, id) => {
    if (role === 'admin') {
      this.props.fetchALLSessions();
    } else if (role === 'trainer') {
      this.props.fetchTrainerSessions(id);
    } else if (role === 'localLead') {
      this.props.fetchLocalLeadSessions(id);
    }
  };

  clickToggle = direction => {
    this.setState({ toggle: direction });
  };

  render() {
    const { headerTitle, toggle } = this.state;

    const { sessions } = this.props;
    if (!sessions) {
      return <div>loading</div>;
    }
    const upcomingSessions =
      sessions &&
      sessions.filter(item => {
        return moment(item.date).diff(moment(), 'seconds') > 0;
      });

    const selectedSessions = toggle === 'left' ? sessions : upcomingSessions;

    const sortedSessions = selectedSessions.sort(
      (a, b) => moment(b.date).valueOf() - moment(a.date).valueOf()
    );

    return (
      <ViewSessionsWrapper>
        <Header label={headerTitle} type="section" />
        <TotalSessions>
          <Span>
            {toggle === 'left' ? 'All Sessions:' : 'Upcoming Sessions:'}
          </Span>
          <SessionsCount>
            {selectedSessions && selectedSessions.length}
          </SessionsCount>
        </TotalSessions>
        <HeaderSection>
          <Toggle
            large
            leftText="All Sessions"
            rightText="Upcoming sessions"
            selected={toggle}
            onClick={this.clickToggle}
          />
        </HeaderSection>
        <LinkBtn to="/create-session">Add New Session</LinkBtn>
        {selectedSessions && selectedSessions.length ? (
          <SessionList dataList={sortedSessions} />
        ) : (
          <Empty description="No Sessions!" style={{ marginTop: '3rem' }} />
        )}
      </ViewSessionsWrapper>
    );
  }
}

const mapStateToProps = state => {
  return {
    userId: state.auth.id,
    role: state.auth.role,
    sessions: state.sessions.sessions,
  };
};

export default connect(
  mapStateToProps,
  { fetchTrainerSessions, fetchLocalLeadSessions, fetchALLSessions }
)(ViewSessions);
