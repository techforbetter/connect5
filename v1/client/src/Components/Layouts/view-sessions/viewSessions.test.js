import React from "react";
import renderer from "react-test-renderer";
import { StaticRouter } from "react-router-dom";

import ViewSessions from "./index";

test("Header matches snapshot", () => {
  const context = {};
  const component = renderer.create(
    <StaticRouter context={context}>
      <ViewSessions />
    </StaticRouter>,
  );
  expect(component).toMatchSnapshot();
});
