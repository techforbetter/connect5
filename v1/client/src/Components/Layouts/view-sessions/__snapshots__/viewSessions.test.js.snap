// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header matches snapshot 1`] = `
<div
  className="sc-bdVaJa duKoVA"
>
  <div
    className="sc-bwzfXH eItADb"
  >
    <div
      className="sc-htpNat kafyHU"
    >
      <h1
        className="sc-bxivhb eVLUcy"
      >
        Sessions
      </h1>
    </div>
  </div>
  <div
    className="sc-ifAKCX ibWFwJ"
  >
    <div
      className="sc-EHOje cbnCUa"
    >
      <div
        className="sc-bZQynM ebTKbK"
      >
        <div
          className="sc-htoDjs kCnhJU"
        >
          Date
        </div>
        <div
          className="sc-htoDjs kCnhJU"
        >
          Type
        </div>
        <div
          className="sc-htoDjs kCnhJU"
        >
          Details
        </div>
      </div>
    </div>
    <div
      className="sc-dnqmqq fqNGUe"
    />
  </div>
</div>
`;
